src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -110,134 +113,68 @@ public ArmSubsystem() {
         }
 
         // Set default PID values elbow
-        for (var config : new PID[]{ELBOW_DEFAULT, ELBOW_STEADY}) {
+        for (var config : new PID[] {ELBOW_DEFAULT, ELBOW_STEADY}) {
             m_elbow.getPIDController().setP(config.P, config.slot);
             m_elbow.getPIDController().setI(config.I, config.slot);
             m_elbow.getPIDController().setD(config.D, config.slot);
             m_elbow.getPIDController().setFF(config.FF, config.slot);
             m_elbow.getPIDController().setDFilter(config.DF, config.slot);
         }
 
+        m_elbow.enableSoftLimit(SoftLimitDirection.kForward, true);
+        m_elbow.enableSoftLimit(SoftLimitDirection.kReverse, true);
+        m_shoulder.enableSoftLimit(SoftLimitDirection.kForward, true);
+        m_shoulder.enableSoftLimit(SoftLimitDirection.kReverse, true);
+
         // Set the soft limits
         m_shoulder.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, 60);
         m_shoulder.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, -15);
         m_elbow.setSoftLimit(CANSparkMax.SoftLimitDirection.kForward, 60);
         m_elbow.setSoftLimit(CANSparkMax.SoftLimitDirection.kReverse, -15);
+    }
 
+    public void setState(State desiredState) {
+        this.m_desiredState = desiredState;
     }
 
-    public void setState(State state) {

```

Comment:
This whole function is required.  Don't change it. There are limitations on what state you can reach from another state. This is to protect the driver from accidentally extending after picking up.

Reply:
Right... But what if the driver *wants* to go from pickup to extended? I understand that you want to make a definition state machine, but it doesn't make sense to limit the driver's functionality. If there's a piece right in front of the goal it doesn't make sense for the driver to go `Pick up -> rest -> place`. Going `pick up -> place` will shave a second or so off the time.

Reply:
So the reasoning for this is that the driver still has that capability. When you're picking up you'll almost always be in a location or orientation that does not beget a clear transition to the extended positions. There is a massive danger with extending the arm while transiting the field or being around other robots. So if we accidentally extend the consequences are quite dramatic. This is also likely as it is very common especially at the first competition for drivers to hit the wrong button. Doing so in a less than ideal location could have one of a few consequences; fouling by hitting and entering another robot; tipping us; damaging the arm etc. The consequence of implementing this restriction is about an extra second to wait for the transition to Rest to complete. This can be done while the driver is moving safely.

If you think this doesn't make sense feel free to disagree just keep in mind how frequently each scenario will be happening and the risk associated with incorrectly extending the arm in each scenario.

Reply:
I do disagree because I see the driver picking up game pieces less than a meter from the station pretty often. So he'd save time going straight from the pick up state to the placing state. I'll implement this anyway though.